test/teiserver/game/balancer_server_test.exs

```diff
@@ -174,4 +174,46 @@ defmodule Teiserver.Game.BalancerServerTest do
     poll_until_nil(fn -> GenServer.call(pid, :get_current_balance) end)
     assert Map.get(GenServer.call(pid, :report_state), :last_balance_hash, :not_found) == nil
   end
+
+  test "load 250 players in and run a balance pass or three, for perf testing" do
+    team_count = 2
+
+    players =
+      make_users_with_ranks(Enum.to_list(1..250))
```

Comment:
this isn't going to trigger the problematic behaviour, that test doesn't bring much value.
If you want to trigger the memory explosion, you need to have some groups of users in parties, and some without.